File: app/schemas/consents.py
from datetime import datetime

from pydantic import BaseModel


class ConsentStatusResponse(BaseModel):
    required: bool
    accepted: bool
    consent_type: str
    current_version: str
    accepted_at: datetime | None = None


class ConsentAcceptRequest(BaseModel):
    consent_type: str
    consent_version: str


class ConsentAcceptResponse(BaseModel):
    accepted: bool
    consent_type: str
    consent_version: str
    accepted_at: datetime

File: frontend/src/types/consents.ts
export interface ConsentStatus {
  required: boolean;
  accepted: boolean;
  consent_type: "disclaimer";
  current_version: string;
  accepted_at: string | null;
}

export interface ConsentAcceptPayload {
  consent_type: "disclaimer";
  consent_version: string;
}

export interface ConsentAcceptResponse {
  accepted: boolean;
  consent_type: "disclaimer";
  consent_version: string;
  accepted_at: string;
}
